python/lsst/obs/base/ingest.py

```diff
@@ -220,10 +232,40 @@ def extractMetadata(self, filename: str) -> RawFileData:
             as well as the original filename.  All fields will be populated,
             but the `RawFileData.dataId` attribute will be a minimal
             (unexpanded) `DataCoordinate` instance.
+
+        Notes
+        -----
+        Assumes that there is a single dataset associated with the given
+        file.  This method must be subclassed if multiple datasets are
+        stored in a single raw file.
         """
         phdu = readMetadata(filename, 0)
         header = merge_headers([phdu, readMetadata(filename)], mode="overwrite")
         fix_header(header)
+        datasets = [self._calculate_dataset_info(header, filename)]
+
+        # Assume all datasets in this file use the same Formatter
+        FormatterClass = self.instrument.getRawFormatter(datasets[0].dataId)
```

Comment:
The comment seems redundant since, as implemented, there is only one dataset in this file.

Reply:
The comment is for someone who is trying to work out how to write their own extractMetadata.